Refactor to add QCMetric class and list of URLs for reference images/figures

dbirman opened the pull request.

saskiad commented on examples/quality_control.json:
```diff
@@ -12,11 +12,18 @@
+            {
+               "name": "Video_frame_match",
+               "value": {
+                  "Video_1_num_frames": 662,
```
I guess I thought that each metrics would be a separate instance of `qc_metrics`, is that not the idea?

dbirman replied:
Oh I see what you were imagining, so the evaluation here would have two metrics (video 1 frames / video 2 frames) and then the evaluation is passed/failed based on whether they match? (let me know if that's wrong). I'll change the example.

dbirman replied:
For evaluations that have rules like this, do we need a place to put that? I feel like "Notes" doesn't imply that there should be a description of the evaluation rule.

saskiad commented on src/aind_data_schema/core/quality_control.py:
```diff
@@ -19,22 +19,30 @@ class Status(str, Enum):
+    """Description of a single quality control metric"""
+    name: str = Field(..., title="Metric name")
+    value: Any = Field(..., title="Metric value")
+    description: Optional[str] = Field(None, title="Metric description")
```
add the `default=` for the optional fields.

saskiad reviewed (changes requested):
fix the defaults. Also a quick question on how you envision using it

dbirman commented:
Added quality_control.Stage per discussion, and modified examples to better reflect our concept. Also added name to QCEvaluation and made descriptions optional, to separate stage/name/desc from each other.

saskiad commented on src/aind_data_schema/core/quality_control.py:
```diff
@@ -19,25 +19,46 @@ class Status(str, Enum):
-    evaluator_full_name: str = Field(..., title="Evaluator full name")
+    evaluation_stage: Stage = Field(..., title="Evaluation stage")
+    evaluation_name: str = Field(..., title="Evaluation name")
+    evaluation_desc: Optional[str] = Field(default=None, title="Evaluation description")
```
can we make this `evaluation_description` - I'm reluctant to shorten things, even obvious things.

saskiad commented on src/aind_data_schema/core/quality_control.py:
```diff
@@ -19,25 +19,46 @@ class Status(str, Enum):
+    evaluator: str = Field(..., title="Evaluator full name")
     evaluation_date: date = Field(..., title="Evaluation date")
-    qc_metrics: AindGenericType = Field(AindGeneric(), title="QC metrics")
+    qc_metrics: List[QCMetric] = Field(title="QC metrics")
```
I think this needs the `...` in the Field

saskiad reviewed (changes requested):
super minor things, but otherwise I think it's good.
